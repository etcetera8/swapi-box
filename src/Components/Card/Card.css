.Card {
  height: 250px;
  margin: 10px;
  width: 200px;
  padding-right: 7px;
  font-family: 'Fjalla One', sans-serif;
  border: 1px solid #f3da4a;
  box-shadow: 5px 7px rgba(0,0,0, 0.2);
  box-sizing: border-box;
  overflow: scroll;
}


header {
  display: flex;
  justify-content: space-between;
  padding: 0;
  color: #f3da4a;
  letter-spacing: 1px;
  border-bottom: 1px solid #f3da4a;
}

.favorited {
  color: #f3da4a;
}

.name {
  margin: 7px;
}

.favorite {
  margin: 7px;
  max-height: 28px;
  border: 1px solid #f3da4a;
  font-size: 1.2em;
  background: none;
}

.key {
  margin-right: 10px;
  color: #f3da4a;
}

.categories {
  margin: 5px 0 0 0;
  list-style: none;
  padding: 7px;
}

.categories li {
  color: rgba(250, 250, 250, 0.8);
  margin: 0px 0 15px 0 ;
}