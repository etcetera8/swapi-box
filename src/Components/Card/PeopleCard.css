.PeopleCard {
  background: rgba(255, 10, 255, 0.1);
}