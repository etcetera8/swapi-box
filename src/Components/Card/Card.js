import React from 'react';
import PropTypes from 'prop-types';
import './PeopleCard.css';
import './Card.css';

const Card = ({person, setFavorite, className}) => {
  const {name, homeworld, species, population} = person;
  return (
    <article className='Card PeopleCard'> 
      <header>
        <h1 className="name">{name}</h1>
        <button onClick={() => setFavorite(name, 'people')} className={`favorite ${className}`}>â˜…</button>
      </header>
      <ul className="categories">
        <li><span className='key'>Homeworld:</span> {homeworld}</li>
        <li><span className='key'>Species:</span> {species}</li>
        <li><span className='key'>Population:</span> {population}</li>
      </ul> 
    </article>
  )
}

Card.propTypes = {
  person: PropTypes.shape({
    name: PropTypes.string.isRequired,
    homeworld: PropTypes.string.isRequired,
    species: PropTypes.string.isRequired,
    population: PropTypes.string.isRequired,
  }),
  setFavorite: PropTypes.func.isRequired,
  className: PropTypes.string.isRequired
};

export default Card;