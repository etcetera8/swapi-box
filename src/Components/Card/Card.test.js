
import React from 'react';
import ReactDOM from 'react-dom';
import Card from './Card';
import { shallow } from 'enzyme';

describe('Card', () => {
  let wrapper;
  const mockFunction = jest.fn();
  
  beforeEach( () => {
    wrapper = shallow(<Card
                        className={'activepeople'}
                        setFavorite={mockFunction}
                        name={'luke'}
                        species={'human'}
                        homeworld={'naboo'}
                        population={'20'}
                        key={1}
                      />)
  })

  it('should match the snapshot test', () => {
    expect(wrapper).toMatchSnapshot();
  })
})