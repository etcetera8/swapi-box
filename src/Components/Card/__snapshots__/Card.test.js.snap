// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot test 1`] = `
<article
  className="Card"
>
  <header>
    <h1
      className="name"
    >
      luke
    </h1>
    <button
      className="favorite activepeople"
      onClick={[Function]}
    >
      â˜…
    </button>
  </header>
  <ul
    className="categories"
  >
    <li>
      <span
        className="key"
      >
        Homeworld:
      </span>
       
      naboo
    </li>
    <li>
      <span
        className="key"
      >
        Species:
      </span>
       
      human
    </li>
    <li>
      <span
        className="key"
      >
        Population:
      </span>
       
      20
    </li>
  </ul>
</article>
`;
